src/main/kotlin/dev/restate/sdktesting/tests/KillRuntime.kt

```diff
@@ -35,24 +38,30 @@ class KillRuntime {
   @Timeout(value = 60, unit = TimeUnit.SECONDS)
   @Test
   fun startAndKillRuntimeRetainsTheState(
-      @InjectClient ingressClient: Client,
       @InjectContainerHandle(RESTATE_RUNTIME) runtimeHandle: ContainerHandle
   ) = runTest {
-    var counterClient = CounterClient.fromClient(ingressClient, "my-key")
-
-    val res1 = counterClient.add(1, idempotentCallOptions())
+    // We instantiate the client manually, in order to close it before killing and restarting
+    var httpClient = HttpClient.newHttpClient()
+    var ingressClient =
+        DefaultClient.of(
+            httpClient, "http://127.0.0.1:${runtimeHandle.getMappedPort(8080)!!}", HashMap())
+    val res1 = CounterClient.fromClient(ingressClient, "my-key").add(1, idempotentCallOptions())
     assertThat(res1.oldValue).isEqualTo(0)
     assertThat(res1.newValue).isEqualTo(1)
 
+    // Close the HTTP client to avoid keeping around dangling connections.
+    httpClient.close()
+
     // Stop and start the runtime
     runtimeHandle.killAndRestart()
 
     // We need a new client, because on restarts docker might mess up the exposed ports. NotFunky
     // but true...
-    counterClient =
-        CounterClient.fromClient(
-            Client.connect("http://127.0.0.1:${runtimeHandle.getMappedPort(8080)!!}"), "my-key")
-    val res2 = counterClient.add(2, idempotentCallOptions())
+    httpClient = HttpClient.newBuilder().connectTimeout(5.seconds.toJavaDuration()).build()
+    ingressClient =
+        DefaultClient.of(
+            httpClient, "http://127.0.0.1:${runtimeHandle.getMappedPort(8080)!!}", HashMap())
+    val res2 = CounterClient.fromClient(ingressClient, "my-key").add(2, idempotentCallOptions())
```

Comment:
Will this client automatically retry on connect timeouts?

Reply:
I don't think, adding that now